### PR TITLE
Tensorboard callback

### PR DESCRIPTION
**Pull request by ASvyatkovskiy**
This PR implements Tensorboard callback allowing to visualize graphs of the training and test metrics, as well as activation histograms for the different layers in the model.

PR introduces a `Tensorboard` class with custom `set_model`, `on_epoch_end` and `on_train_end` methods.

Currently monitoring: 
1. Scalar variable summaries for training and validation loss, validation ROC, per epoch
1. Weights, biases and activations histograms and distributions, per each layers, per epoch
1. Graph visualization

Note: completely independent on Keras, but consider inheriting from Callback in future. Reason: current Keras implementation does not allow passing validation data as a generator, while loading the whole list causes OOM.

Note: free parameter is the number of validation steps, which is passed along with the validation data generator. Consider changing the value as appropriate

### CODE REVIEW

**Comment by ASvyatkovskiy** [comment]
image
